### PR TITLE
Add initial unit tests (solvers)

### PR DESCRIPTION
**Pull request by RobertPincus**
This PR adds some unit tests for the solvers alone (in `tests/rte[sl]w_unit_tests.F90`). Longwave solvers are tested for correctness in radiative equilibrium and for invariance to choices (vertical orientation, problem size...). Shortwave solvers are tested only for invariance.

### CODE REVIEW

**Comment by RobertPincus** [comment]
Note that the standalone solver tests fail for `nvfortran` 24.1 and the default kernels. This is surprising because the same compiler/kernel combination passes similar tests in the end-to-end problem (`tests/check_equivalence`). @alexeedm has promised to take a look in the coming days.

**Comment by RobertPincus** [comment]
@skosukhin What would you think about merging this onto `develop` despite the failing CI for nvidia+default-kernels? I'll chase that down after the fact.

**Comment by RobertPincus** [comment]
Of course @alexeedm If you have any bandwidth to look at the failing `nvfortran 24.1`/default kernels failures I'm grateful. You had thought there might be a compiler bug.

File: tests/mo_testing_utils.F90 [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    real(wp), optional,       intent(in) :: tol 
+    
+    real(wp) :: tolerance 
+    if (present(tol)) then 
+      tolerance = tol 
+    else
+      tolerance = 2._wp
+    end if 
+
+    allclose_2= all(abs(array1-array2) <= tolerance * spacing(array1))
```
Comment (skosukhin) [inline tests/mo_testing_utils.F90:69]: I think that `spacing` should be calculated based on the reference data, which is often (I don't know if that's always the case) passed as the second argument.

**Review by skosukhin — CHANGES_REQUESTED** [review]
@alexeedm please, take a look at the suggested workaround for NVHPC. It looks like the `-Mbounds` introduces a bug into the generated code.

**Review by skosukhin — APPROVED** [review]

File: tests/mo_testing_utils.F90 (hunk 1 above)
Comment (RobertPincus) [reply tests/mo_testing_utils.F90]: Made a more sweeping change in b0ca961 to clarify reference and test values and to use spacing from the ref values

**Pull request merged**